components/icu4x_ecma402/src/pluralrules.rs

```diff
@@ -0,0 +1,340 @@
+                    opts: opt(3, 2, 3, 3, 4),
+                    expected: "000.00",
+                },
+                TestCase {
+                    n: 1.5,
+                    opts: opt(3, 2, 3, 3, 4),
+                    expected: "001.50",
+                },
+                TestCase {
+                    n: 1.5,
+                    opts: opt(5, 5, 8, 3, 4),
+                    expected: "00001.50000",
+                },
+                TestCase {
+                    n: 123456.5,
+                    opts: opt(5, 5, 8, 3, 4),
+                    expected: "123400.00000",
+                },
+                TestCase {
+                    n: 123456.5432112345,
+                    opts: opt(5, 5, 8, 3, 4),
+                    expected: "123400.00000000",
+                },
+            ];
+            for test in tests {
+                let actual = fixed_format(test.n, &test.opts);
+                assert_eq!(test.expected, actual.as_str(), "test: {:?}", &test);
+            }
+        }
+
+        #[test]
+        fn format_conversion() {
+            #[derive(Debug)]
+            struct TestCase {
+                n: f64,
+                opts: Options,
+                expected: PluralOperands,
+            };
+            let tests = vec![TestCase {
+                n: 1.5,
+                opts: Options {
+                    in_type: Type::Cardinal,
+                    minimum_integer_digits: 3,
+                    minimum_fraction_digits: 2,
+                    maximum_fraction_digits: 3,
+                    minimum_significant_digits: 3,
+                    maximum_significant_digits: 4,
+                },
+                expected: PluralOperands {
+                    i: 1,
+                    v: 2,
+                    w: 1,
+                    f: 50,
+                    t: 5,
+                },
+            }];
+            for test in tests {
+                let actual = to_icu4x_operands(test.n, test.opts.clone());
+                assert_eq!(test.expected, actual, "test: {:?}", &test);
+            }
+        }
+    }
+}
+
+pub struct PluralRules {
+    opts: ecma402_traits::pluralrules::Options,
+    rep: ipr::PluralRules,
+}
+
+impl ecma402_traits::pluralrules::PluralRules for PluralRules {
+    type Error = PluralRulesError;
+
+    fn try_new<L>(l: L, opts: ecma402_traits::pluralrules::Options) -> Result<Self, Self::Error>
+    where
+        L: ecma402_traits::Locale,
+        Self: Sized,
+    {
+        // TODO: introduce a global data provider here.
+        let dp = icu_data_provider::InvariantDataProvider;
+        PluralRules::try_new_with_provider(l, opts, &dp)
+    }
+
+    fn select<W>(&self, number: f64, writer: &mut W) -> std::fmt::Result
+    where
+        W: std::fmt::Write,
+    {
+        let op = internal::to_icu4x_operands(number, self.opts.clone());
+        let result = self.rep.select(op);
+        write!(writer, "{}", internal::as_str(result))
+    }
```

Comment:
Hmm, I'm rethinking this :( I'm sorry.

Since PluralRules are not intended to be stringified and categories are going to be used as selections, the writer trait here seems to be extraneous. I still believe we should do this for formatters, but maybe here it makes sense to have `select() -> &str` since it returns a category?

Reply:
We can certainly add a selector that returns `&'static str`.